features/facebookImport/src/context/importer-context.jsx

```diff
@@ -234,11 +237,22 @@ export const ImporterProvider = ({ children }) => {
     //on file change
     //when files changed run the importer first and create an account model first.
     //after there is an account the analyses are triggered.
-    useEffect(() => {
-        if (files?.[0])
-            importData(files[0]).then((newFacebookAccount) =>
-                setFacebookAccount(newFacebookAccount)
-            );
+    useEffect(async () => {
```

Comment:
The use of cache would be easier to handle at the storage level; it should be transparent at this level. Also, why the change to `async`? What's going to be the effect for those using this class?